### DISCUSSION TITLE
ReactionType with name `Like` not exists.

### DISCUSSION THREAD

**Discussion by alquialrium** [cybercog/laravel-love#212] [Choose solution] [answered]
Hello, I'm starting use de package, I'm developing a small twitter clone, now trying to do like or dislike to each tweet but I'm having troubbles.

My Like Button

    `{!! Form::open(['route' => 'answers.like', 'class' => 'd-inline transparent']) !!}
     {{ Form::hidden('tweet_id', $tweet->id) }}
     {{ Form::hidden('user_id', auth()->user()->id) }}
         <button type="submit" title="{{ __('I like') }}" class="fs-5 d-inline nav-link text-gray-500 fw-light" href="#">
             <i class="far fa-heart"></i>
          </button>
     {!! Form::close() !!}
     `

My Route

    `Route::post('tweets/like', [App\Http\Controllers\TweetController::class, 'like'])->name('tweets.like');`

My Controller

    `public function like(Request $request)
    {
        $user = User::findOrFail($request->user_id);
        $tweet = Tweet::findOrFail($request->tweet_id);

        $user->viaLoveReacter()->reactTo($tweet, 'Like');
    }`

My Tweet Model

    `namespace App\Models;

    use Illuminate\Database\Eloquent\Factories\HasFactory;
    use Illuminate\Database\Eloquent\Model;

    use Cog\Contracts\Love\Reactable\Models\Reactable as ReactableInterface;
    use Cog\Laravel\Love\Reactable\Models\Traits\Reactable;

   class Tweet extends Model implements ReactableInterface
   {
       use HasFactory, Reactable;

       protected $fillable = [
            'user_id',
            'tweet',
            'photo',
            'likes',
            'status',
    ];`

My User Model

    `namespace App\Models;

    use Illuminate\Support\Facades\DB;
    use Illuminate\Contracts\Auth\MustVerifyEmail;
    use Illuminate\Database\Eloquent\Factories\HasFactory;
    use Illuminate\Foundation\Auth\User as Authenticatable;
    use Illuminate\Notifications\Notifiable;
    use Laravel\Sanctum\HasApiTokens;

    use Overtrue\LaravelFollow\Traits\Follower;
    use Overtrue\LaravelFollow\Traits\Followable;

    use Cog\Contracts\Love\Reacterable\Models\Reacterable as ReacterableInterface;
    use Cog\Laravel\Love\Reacterable\Models\Traits\Reacterable;

    class User extends Authenticatable implements MustVerifyEmail, ReacterableInterface
   {
        use HasApiTokens, HasFactory, Notifiable, Follower, Followable, Reacterable;

        protected $fillable = [
            'first_name',
            'last_name',
            'username',
           'email',
           'password',
    ];`

I have that error:ReactionType with name `Like` not exists

**Comment by antonkomarev** [comment, marked as answer]
Hi, @alquialrium! Have you created all reaction types you need?
https://laravel-love.readme.io/docs/manage-reaction-types

**Comment by alquialrium** [comment]
I had executed this command `php artisan love:reaction-type-add --default` but for some reason it wasn't working, I executed it again and it already does the record in the database. Thank you

**Reply by antonkomarev** [reply]
@alquialrium This command creates Like and Dislike reaction types. If you allowing only likes, you can drop Dislike reaction type.

**Reply by alquialrium** [reply]
Thank you!

**Comment by alquialrium** [comment]
Another question, when I run `php artisan miagrate:fresh --seed` I have to run the command again `php artisan love:reaction-type-add --default`?

**Reply by antonkomarev** [reply]
Yes, after refreshing database you have to add reaction types because they are stored in database.

You can create your own seeder which will create reaction types you need, that [command](https://github.com/cybercog/laravel-love/blob/80c7ad2f16b56b51ca3eeb5098f24647e7f0d913/src/Console/Commands/ReactionTypeAdd.php) is pretty simple.
